core/base/src/TSystem.cxx

```diff
@@ -2865,9 +2865,9 @@ int TSystem::CompileMacro(const char *filename, Option_t *opt,
    if (mode==kDefault) {
       TString rootbuild = ROOTBUILD;
       if (rootbuild.Index("debug",0,TString::kIgnoreCase)==kNPOS) {
-         mode=kOpt;
+         mode |= kOpt;
```

Comment:
In this section mode is zero (kDefault) so the new code is strictly equivalent to the old code just ... more confusing.

Reply:
Imho the new version does a good job at expressing that `mode` can (in general) be a combination of values rather than one specific value. `mode=kOpt` will more easily become wrong after refactoring.

Reply:
Except that *unless* you go back and look at the value of kDefault, the reader of *this* portion of the code can be distracted in thinking that *here* it is a combination while *here* it is actually *explicitly* an exact assignment (knowing that the two are here the same requires more thought and searches through the header file)

Reply:
Agreed!